src/app/components/ContactPgpSettings.js

```diff
@@ -42,6 +42,13 @@ const ContactPgpSettings = ({ model, setModel }) => {
 
         await Promise.all(
             files.map(async (publicKey) => {
+                if (!publicKey.isPublic()) {
+                    // do not allow to upload private keys
+                    return createNotification({
+                        type: 'error',
+                        text: c('Error').t`Invalid public key file`
+                    });
+                }
```

Comment:
In case `createNotification` return something different, you should return a `Promise`:

``` js
createNotification({
    type: 'error',
    text: c('Error').t`Invalid public key file`
});
return Promise.resolve();
```

Reply:
But isn't the return of `createNotification` already wrapped inside a promise because it's inside an `async` function?

Reply:
I was not aware that `createNotification` was returning a `Promise`. Since it's not something obvious, it may change in the future. You can keep your logic as it is.